### ISSUE TITLE
Migration to v6 from v5 basic questions

### ISSUE THREAD

**Issue by brotzky** [erikras/redux-form#967]
Hi, I'm working on migrating from v5 to v6 and have some questions. 

1. Within the new Field what is the best way to handle onChange? In v5 I had ```<input onChange={this.func()}  {...fieldName} />```, but that seems to break for me with Field wrapper.

2. What's the best way to use field values within a component class? In v5, with the fields array, I would often go ```const { fields: { inputField } } = this.props;```at the top of a method and could then use inputField.value, for example. With the new Fields way are we supposed to mapStateToProps and then pull the value directly off props?

what I often did in v5 
```
 parseNumber() {
    const { fields: { inputField } } = this.props;
    ...
  }
```

In the documentation, I think there may be a small typo in the examples for migrating from v5 to v6. It may have been on purpose, but confused me initially. 
http://redux-form.com/6.0.0-alpha.8/docs/MigrationGuide.md/
```
export default reduxForm({
  name: 'myForm' // <-- Is this supposed to be **form**: 'myForm' ?
})(MyForm)
```
http://redux-form.com/6.0.0-alpha.8/examples/simple/
```
export default reduxForm({
  form: 'simple'  // a unique identifier for this form
})(SimpleForm)
```

Thanks guys. Migrating looked straight forward, but.. 🙈

**Comment by erikras** [comment]
Right on, on the typo. Thanks for that. It's so easy to make mistakes in documentation code, because you don't actually run it. This is why I took great pains to be sure that the code shown on the examples pages _is actually the same code that is running_.

To your earlier questions...

1. That specific form would not have worked in `v5` either, because the subsequent destructuring of `{...fieldName}` into the `<input>` will overwrite the previous `onChange` prop. I keep meaning to write up a doc page about prop destructuring, because the React documentation is pretty light about it, and it causes headaches.
2. Yes, this is a fairly big difference, and probably the most annoying thing about moving from `v5` to `v6`. You are exactly right that the answer is to `connect()` to the form values manually. However, I'm planning on making a selector to make that more convenient. Something like:

```js
import { reduxForm, formValue } from 'redux-form'

@connect(
  state => ({
    aValue: formValue(state, 'myForm', 'aValue'),
    bValue: formValue(state, 'myForm', 'even.deeper.bValue'),
  })
)
@reduxForm({
  form: 'myForm'
})
class MyForm extends Component {
  calculateSomething() {
    const { aValue, bValue } = this.props
    return aValue + bValue
  }
 
  render() {
  }
}
```

I haven't figured out the exact syntax, but you're welcome to provide suggestions.

**Comment by brotzky** [comment]
Thanks!

Interesting point about onChange. What would be the best way to handle onChange in a Field input? You don't have access to the redux-form/CHANGE action directly and can't handle it within the component? Maybe I'm still missing something.

I don't have any solid suggestions off the top of my head for connecting the form values. It's nice if you have split your components up into small pieces where you have to connect only one field per component. I can see it being annoying if you have your entire form in a single component and you need to connect a ton of field values for some reason. It's worth noting that connect() is already a simple solution that people would understand without having to learn something new.

**Comment by erikras** [comment]
> What would be the best way to handle `onChange` in a `Field` input? 

You're overthinking it. As long as you call the provided `onChange` function, you can do whatever else you want.

```js
<Field name="myField" component={props =>
  <input
    {...props}
    onChange={event => { // <---- remember this will overwrite the default one provided in props
      props.onChange(event) // call the provided one to update the redux state
      const value = event.target.value
      doWhateverWeWant('myField', value)
    }}
  />
}/>
```

>  I have run into a strange issue where I am trying to use `connect()`, but the field value doesn't exist until someone focuses it. Thus, when I am trying to connect it fails as I am trying to connect a value that doesn't exist yet in the state.

Yep, this is #945. You've got to do null checks at every stage. e.g.

```js
@connect(
  state => ({
    myFieldValue: state.form && state.form.myForm && state.form.myForm.values && state.form.myForm.values.myField
  })
)
```

Pretty disgusting, hence my desire to write a helper function.

**Comment by brotzky** [comment]
Thanks @erikras 

Agreed on the helper function

**Comment by brotzky** [comment]
Another issue I've run into.

When I type in an input field it appears to go out of focus after each stroke. I have to click back into the field to type another character. This issue only occurred after moving to v6.

**Comment by erikras** [comment]
> When I type in an input field it appears to go out of focus after each stroke.

Known bug #961.

**Comment by erikras** [comment]
Published my "form value selector" API as [`v6.0.0-alpha.11`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-alpha.11). Feedback welcome!
